test/sys/test_wait.rs

```diff
@@ -56,6 +56,61 @@ fn test_waitstatus_pid() {
     }
 }
 
+#[test]
+fn test_child_iterator() {
+    let _m = ::FORK_MTX
+        .lock()
+        .expect("Mutex got poisoned by another test");
+
+    let mut children = Vec::new();
+
+    // create some child_processes and kill them immediatly so we can collect the exit codes
+    for _ in 0..10 {
+        // Safe: The child only calls `pause` and/or `_exit`, which are async-signal-safe.
+        match fork().expect("Error: Fork Failed") {
+            Child => {
+                pause();
+                unsafe { _exit(123) }
+            }
+            Parent { child } => {
+                kill(child, Some(SIGKILL)).expect("Error: Kill Failed");
+                children.push(child);
+            }
+        }
+    }
+
+    let start_time = std::time::Instant::now();
+
+    let mut exited_children = Vec::new();
+    
+    // this runs until all children have been signaled. It might have to call the iterator a few times, because the events might
+    // be delivered in chunks
+    while exited_children.len() < children.len() {
+        // this prevents the test from running for ever in case of a bug 
+        assert!(
+            start_time.elapsed() < std::time::Duration::from_secs(2), 
+            "It takes the children way too long to exit, something is probably broken"
+        );
+        exited_children.extend(child_event_iter());
```

Comment:
If this is a polling loop, shouldn't it sleep somewhere?  Also, the comment on line 86 is >> 80 cols.

Reply:
It is not really necessary to sleep here, the children exit immediately and are collected yb the loop. Sleeping would in most cases increase the time this tests needs.

At worst (when a bug occurs) this will occupy one cpu core for 2 seconds.